src/vm-repair/azext_vm_repair/_validators.py

```diff
@@ -0,0 +1,199 @@
+        try:
+            find_repair_command = 'az resource list --tag {tag} --query "[?type==\'Microsoft.Compute/virtualMachines\']"' \
+                                  .format(tag=tag)
+            logger.info('Searching for repair-vm within subscription:')
+            output = _call_az_command(find_repair_command)
+        except AzCommandError as azCommandError:
+            logger.error(azCommandError)
+            raise CLIError('Unexpected error occured while locating repair VM.')
+        repair_list = loads(output)
+
+        # No repair VM found
+        if not repair_list:
+            raise CLIError('Repair VM not found for {vm_name}. Please check if the repair resources were removed.'.format(vm_name=namespace.vm_name))
+
+        # More than one repair VM found
+        if len(repair_list) > 1:
+            message = 'More than one repair VM found:\n'
+            for vm in repair_list:
+                message += vm['id'] + '\n'
+            message += '\nPlease specify the --repair-vm-id to restore the disk-swap with.'
+            raise CLIError(message)
+
+        # One repair VM found
+        namespace.repair_vm_id = repair_list[0]['id']
+        logger.warning('Repair-vm-id not given, restoring from only matching repair VM: %s', namespace.repair_vm_id)
+
+    if not is_valid_resource_id(namespace.repair_vm_id):
+        raise CLIError('Repair resource id is not valid.')
+
+    repair_vm_id = parse_resource_id(namespace.repair_vm_id)
+    # Check if data disk exists on repair VM
+    repair_vm = get_vm(cmd, repair_vm_id['resource_group'], repair_vm_id['name'])
+    data_disks = repair_vm.storage_profile.data_disks
+    if not data_disks:
+        raise CLIError('No data disks found on repair VM: {}'.format(repair_vm_id['name']))
+
+    # Populate disk name
+    if not namespace.disk_name:
+        namespace.disk_name = data_disks[0].name
+        logger.warning('Disk-name not given. Defaulting to the first data disk attached to the repair VM: %s', data_disks[0].name)
+    else: # check disk name
+        if not [disk for disk in data_disks if disk.name == namespace.disk_name]:
+            raise CLIError('No data disks found on the repair VM: \'{vm}\' with the disk name: \'{disk}\''.format(vm=repair_vm_id['name'], disk=namespace.disk_name))
+
+def _prompt_repair_username(namespace):
+
+    from knack.prompting import prompt, NoTTYException
+    try:
+        namespace.repair_username = prompt('Repair VM Admin username: ')
+    except NoTTYException:
+        raise CLIError('Please specify username in non-interactive mode.')
+
+def _prompt_repair_password(namespace):
+    from knack.prompting import prompt_pass, NoTTYException
+    try:
+        namespace.repair_password = prompt_pass('Repair VM Admin Password: ', confirm=True)
+    except NoTTYException:
+        raise CLIError('Please specify password in non-interactive mode.')
+
+def _classic_vm_exists(cmd, resource_group_name, vm_name):
+    api_version = '2017-04-01'
```

Comment:
Consider moving these constants to a more visible place if there's a need to update these in the future.

Reply:
Yeah I should do that or have a more reliable method to get the supported api_version. Will look into this. Thanks.